tests/system/projects/test_project.py

```diff
@@ -984,3 +1016,40 @@ def test_project_build_config_export_import(self):
         assert run_result.output("score")
 
         shutil.rmtree(project_dir, ignore_errors=True)
+
+    @pytest.mark.parametrize(
+        "name, save_secrets, expected_states",
+        [
+            (
+                "load-project-secrets",
+                True,
+                [
+                    mlrun.common.schemas.BackgroundTaskState.running,
+                    mlrun.common.schemas.BackgroundTaskState.succeeded,
+                ],
+            ),
+            (
+                "load-project-secrets-1",
+                False,
+                [mlrun.common.schemas.BackgroundTaskState.succeeded],
+            ),
+        ],
+    )
+    def test_load_project_remotely_with_secrets(
```

Comment:
I would add a scenario where the actual `load_project` fails, and verify that there are no leftovers after it (maybe in a different test) - can maybe do that by providing a bad `url` to the `load_project`?